core-bundle/src/Security/Logout/LogoutHandler.php

```diff
@@ -47,7 +47,7 @@ public function __construct(ContaoFramework $framework, LoggerInterface $logger
         $this->logger = $logger;
     }
 
-    public function logout(Request $request, Response $response, TokenInterface $token): void
+    public function logout(Request $request, ?Response $response, TokenInterface $token): void
```

Comment:
Does this method not have to follow the interface: https://github.com/symfony/security-http/blob/5.x/Logout/LogoutHandlerInterface.php#L32 ?

Reply:
Hm, interesting. PHP did not complain during testing 🤔 . Then I suppose the real error is that the `LogoutEvent` is missing a response.

Reply:
To me this looks like an error in Symfony. `LogoutEvent::getResponse()` returns either a `Response` object or `null`. However, [`LegacyLogoutHandlerListener::onLogout`](https://github.com/symfony/symfony/blob/252f85c2c249da7d8f2a490b57cc71199d51fcc9/src/Symfony/Bundle/SecurityBundle/Security/LegacyLogoutHandlerListener.php#L41-L43) relies on this method always returning a `Response` object, since the registered `logoutHandler` is supposed to implement `LogoutHandlerInterface` where the second parameter cannot be null (although apparently it can 🤷).

Reply:
could `public function logout(Request $request, Response $response = null, TokenInterface $token): void` be a solution?

Reply:
Both `?Response $resopnse` and `Response $response = null` are actually valid: https://3v4l.org/R45Tn

Reply:
Looks good to me then! 🎉